cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -111,6 +111,11 @@ if (NOT CONFIG_NEWLIB_LIBC AND
   set_compiler_property(APPEND PROPERTY nostdinc_include ${NOSTDINC})
 endif()
 
+if (CONFIG_PICOLIBC AND
+    NOT CONFIG_PICOLIBC_IO_FLOAT)
```

Comment:
setting of a compiler flag should not be dependent on a kconfig setting.
This is a valid flag for gcc, regardless of picolibc.

The decission whether the flag should be used during compilation should generally be taken based on Kconfig, but that is to be done elsewhere, like you already do here:
https://github.com/zephyrproject-rtos/zephyr/blob/0459bb2aca6be054c00bab07a5def195ab4d232e/CMakeLists.txt#L272-L279

Of course the current `if(<condition>)` might need to be adjusted according to the needs.